### DISCUSSION TITLE
Model Exit when implementing stochastic scenarios

### DISCUSSION THREAD

**Discussion by JaKehrl** [spine-tools/SpineOpt.jl#652] [Support]
Hi, 

we are currently implementing an optimization model reflecting a hydropower plant with a reservoir. The water inflow into the reservoir is seasonal and subject to uncertainties for the future. Therefore, it is not certain how large the power plant should be built (investment decision).

Our goal is to determine the optimal scale of the hydropower plant taking into account the different water inflow scenarios, respectively a timeseries of 1 year for a dry/average/wet year.

Our methodology follows the implementation of the stochastic framework described in the SpineOpt user guide. We consider the  3 time series of the predicted water inflow scenarios.

We have attached 3 slides showing the general problem and our implementation steps. Maybe we could get some help because our model always runs into an error. 

Thanks in advance! 

![image](https://github.com/spine-tools/Spine-Toolbox/assets/124700754/017211d9-bc07-49c3-b649-3cf2154469f6)

![image](https://github.com/spine-tools/Spine-Toolbox/assets/124700754/b942556b-96e7-4efb-ba86-623699107e14)

![image](https://github.com/spine-tools/Spine-Toolbox/assets/124700754/024dbe7a-efd3-4608-a4c8-774bc1a4c56d)
[Stochastic Tree.pdf](https://github.com/spine-tools/Spine-Toolbox/files/11581426/Stochastic.Tree.pdf)

**Comment by ptsavol** [comment]
What is the error? Toolbox crashes? Error from SpineOpt.jl in Toolbox's Julia Console? General traceback in Toolbox's console?

**Reply by JaKehrl** [reply]
Hi, 
please find .txt, .err and .out file attached pasted in a .doc file since Github doesn´t allow the other file types. 
It seems like it has something to do with the Date/Time of the time series. However, I am not certainly sure how to interpret this. 

[Output.docx](https://github.com/spine-tools/Spine-Toolbox/files/11589462/Output.docx)

**Comment by soininen** [comment]
I took the liberty to move this discussion to SpineOpt support forums. Hopefully you'll get the issues resolved!

**Comment by DillonJ** [comment]
@JaKehrl if you can provide more details about the error you are getting we would be happy to help as @PekkaSavolainen suggests

**Reply by ptsavol** [reply]
@DillonJ The error is in `output.docx` above. I guess it boils down to this but can't say for sure.

```
Creating stochastic structure...MethodError: no method matching +(::Dates.DateTime, ::Symbol)
```

**Reply by DillonJ** [reply]
Yeah, sorry, just saw that now.

@JaKehrl it looks like a parameter that should be a duration has been entered as a symbol, maybe the roll_forward parameter. Duration parameters should be entered using the parameter value editor and specifying the value as a duration type as below:

![image](https://github.com/spine-tools/SpineOpt.jl/assets/7080191/fe3c0870-9ff2-44a4-b349-5200ef9efdb9)

**Reply by DillonJ** [reply]
By the way, this happens a lot and we should check for it. I created an issue https://github.com/spine-tools/SpineOpt.jl/issues/653

**Reply by DillonJ** [reply]
@JaKehrl you probaly entered a value for `stochastic_scenario_end` as a string (e.g. typing "1D" directly into the parameter value field) when it should have been explicitly entered as a duration parameter using the parameter value editor dialog (as above) accessed via right click -> edit in the parameter value field

**Comment by JaKehrl** [comment]
@DillonJ  Thanks! The problem has been solved. The time series started on the 01-01-2000 instead of the 01-07-2000. Therefore, there was a mismatch. 

However, now we have another problem. Apparently the model is now infeasible what - in our point of view - does not make a lot of sense. We have chosen a dry, average and wet year. 
[Infeasible.pdf](https://github.com/spine-tools/SpineOpt.jl/files/11661151/Infeasible.pdf)
Solving each year by itself, we receive a feasible solution. Distributing probabilities (10% Dry, 80% Average, 10% Wet) causes an infeasible solution.
